### PR TITLE
fix(RELEASE-2389): fail Release on invalid git config

### PR DESCRIPTION
**Pull request by elenagerman**
## Describe your changes
Releases with invalid tenantPipeline git configuration (empty URL/revision or non-existent branch) were stuck "In Progress" indefinitely because git resolution errors weren't recognized as permanent failures.

This fix extends IsRetryableCreationError to classify git configuration errors as non-retriable, leveraging the existing handlePipelineCreationError machinery from [#1398](https://github.com/konflux-ci/release-service/pull/1398)  ([KONFLUX-12127](https://redhat.atlassian.net/browse/KONFLUX-12127)). Now these releases fail immediately with a clear error message instead of retrying forever.

## Relevant Jira
[RELEASE-2389](https://redhat.atlassian.net/browse/RELEASE-2389)

[KONFLUX-12127]: https://redhat.atlassian.net/browse/KONFLUX-12127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RELEASE-2389]: https://redhat.atlassian.net/browse/RELEASE-2389?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: git/references.go [hunk 1]
```diff
@@ -48,7 +57,7 @@ func isRateLimitError(err error) bool {
 // ResolveBranchToSHA resolves a git branch reference to a commit SHA using go-git
 func ResolveBranchToSHA(repoURL, revision string) (string, error) {
 	if repoURL == "" || revision == "" {
```
Comment (theflockers) [inline git/references.go:78]: Maybe also check pathInRepo? The test has `url`, `revision` and `pathInRepo` empty, although we  don't test pathInRepo here.

Comment (elenagerman) [reply git/references.go:78]: good point!
added

**Review by theflockers — DISMISSED** [review]

**Comment by johnbieren** [comment]
Can you create a short demo showing how it was broken before and how it looks now with this fix?

File: controllers/release/adapter_test.go [hunk 2]
```diff
@@ -2226,6 +2226,47 @@ var _ = Describe("Release adapter", Ordered, func() {
...
+			parameterizedPipeline := tektonutils.ParameterizedPipeline{}
+			parameterizedPipeline.PipelineRef = tektonutils.PipelineRef{
+				Resolver: "git",
+				Params: []tektonutils.Param{
+					{Name: "url", Value: ""},
+					{Name: "revision", Value: ""},
+					{Name: "pathInRepo", Value: ""},
+				},
+			}
+			localReleasePlan := &v1alpha1.ReleasePlan{
```
Comment (johnbieren) [inline controllers/release/adapter_test.go]: nit - instead of creating a new one, you can just do a `.deepcopy()` and change the field you need to (in this case `TenantPipeline`)

File: e2e-tests/tests/release/testdata.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ var (
 	LabelNegative                            = ginkgo.Label("release-service", "release-neg", "negMissingReleasePlan")
 	LabelNegBlockReleases                    = ginkgo.Label("release-service", "release-neg", "negBlockReleases")
 	LabelNegManagedPipelineRunCreationDenied = ginkgo.Label("release-service", "release-neg", "negManagedPipelineRunCreationDenied")
+	LabelNegInvalidGitConfig                 = ginkgo.Label("release-service", "release-neg", "negInvalidGitConfig")
```
Comment (johnbieren) [inline e2e-tests/tests/release/testdata.go]: Can we call this like LabelNegTenantPipelineInvalidGitRef or something? Doesn't have to be that, but including at least `Tenant` in it would go a long way IMO

File: e2e-tests/README.md [hunk 4]
```diff
@@ -79,6 +79,7 @@ make test-e2e LABEL=negManagedPipelineRunCreationDenied
 | `release_plan_and_admission` | ReleasePlan ↔ ReleasePlanAdmission matching |
 | `release-neg` | Negative/error scenarios |
 | `negManagedPipelineRunCreationDenied` | Managed release `PipelineRun` create denied by quota; failure is surfaced on `Release` status |
+| `negInvalidGitConfig` | Invalid git resolver config (empty URL/revision); failure is surfaced on `Release` status |
```
Comment (johnbieren) [inline e2e-tests/README.md]: this should mention tenant pipeline

File: git/references.go [hunk 5]
```diff
@@ -27,6 +28,14 @@ import (
 	"github.com/go-git/go-git/v5/plumbing"
 )
 
+// Sentinel errors for git configuration problems that won't resolve on retry.
+var (
+	// ErrInvalidGitConfig indicates the git resolver configuration is invalid (e.g., empty URL or revision).
+	ErrInvalidGitConfig = errors.New("invalid git configuration")
```
Comment (johnbieren) [inline git/references.go]: I think it is worthwhile to include the word `resolver` after `git`

File: loader/helpers_test.go
Comment (johnbieren) [inline loader/helpers_test.go:1]: There is no `loader/helpers.go` so I am not sure why this is adding `loader/helpers_test.go`

Comment (elenagerman) [reply loader/helpers_test.go:1]: Good point. The loader/helpers_test.go was added to provide tests that run without envtest, but it's redundant since loader_test.go already tests the same functions.
removed

File: git/references.go (hunk 5 above)
Comment (elenagerman) [reply git/references.go]: good point!
thank you @johnbieren !
done

File: e2e-tests/README.md (hunk 4 above)
Comment (elenagerman) [reply e2e-tests/README.md]: sure!
done

File: e2e-tests/tests/release/testdata.go (hunk 3 above)
Comment (elenagerman) [reply e2e-tests/tests/release/testdata.go]: sure
changed

File: controllers/release/adapter_test.go (hunk 2 above)
Comment (elenagerman) [reply controllers/release/adapter_test.go]: Just a brilliant idea!
thank you!
modified

**Comment by elenagerman** [comment]
> Can you create a short demo showing how it was broken before and how it looks now with this fix?

sure

**Review by johnbieren — APPROVED** [review]

**Review by twaugh — APPROVED** [review]

**Comment by davidmogar** [comment]
## Production Approval Record

| Field | Value |
|---|---|
| **Action** | APPROVED |
| **Reviewer** | @davidmogar |
| **Timestamp** | 2026-06-24T14:06:05.445Z |
